features/git-town-diff-parent/current_branch/on_non_feature_branch.feature

```diff
@@ -0,0 +1,30 @@
+Feature: git town-diff-parent: errors when trying to diff a perennial branch
+
+  As a developer accidentally trying to diff a perennial branch
+  I should see an error that I cannot diff perennial branches
+  Because perennial branches cannot have parent branches
+
+
+  Background:
+    Given my repository has the perennial branch "qa"
+    And the following commits exist in my repository
```

Comment:
Okay, I will remove the `with local commits` and `with uncommited files`.

Originally, I had left them in to verify that `diff-parent` does not modify the current workspace, but I'm happy to leave these commands out if it will make the tests run faster.

Reply:
The rule of thumb we use is that if the test _changes_ branches or commits, we check that it leaves commits and branches in the expected state. If a command only _reads_ branches and commits, and doesn't change anything, then we don't. 

It's a tradeoff of cost vs precision. The purpose of tests is to increase confidence that Git Town works. The likelihood that a command that only reads things accidentally changes things is already very low.

Reply:
Thanks for the explanation!